sae/consensus.go

```diff
@@ -32,15 +36,89 @@ func (vm *VM) SetPreference(ctx context.Context, id ids.ID, bCtx *block.Context)
 //
 // [accepted]: https://github.com/ava-labs/avalanchego/tree/master/vms#block-statuses
 func (vm *VM) AcceptBlock(ctx context.Context, b *blocks.Block) error {
-	// TODO(arr4n) implement settlement, updating of database canonical block,
-	// head, etc.
-	vm.lastAccepted.Store(b)
-	return vm.exec.Enqueue(ctx, b)
+	// Recall the terminology and ordering from the invariants document:
+	// - (D)isk then (M)emory then (I)nternal then e(X)ternal.
+	// - B accepted after all of B.Settles() settled
+
+	settles := b.Settles()
+	{
+		batch := vm.db.NewBatch()
+
+		// D(s \in S); yes, I know it's in a batch
```

Comment:
IDEs don't render latex inline, so if we want to use the mathematical notation here, I think we should be using the unicode symbols. If we don't want to use the unicode symbols, then I think we should just write it out with words.
```suggestion
		// D(s ∈ S); yes, I know it's in a batch
```
(This suggestion applies to a number of places, I'll just leave it here for you to decide how best to proceed).